static/src/javascripts/projects/common/modules/spacefinder.ts

```diff
@@ -152,15 +153,26 @@ const onInteractivesLoaded = memoize(async (rules: SpacefinderRules) => {
 					}
 
 					const iframe = records[0].addedNodes[0];
-					if (isIframeLoaded(iframe)) {
-						instance.disconnect();
-						resolve();
-					} else {
-						iframe.addEventListener('load', () => {
+
+					// Listen for when the iframes are resized
+					// This is a sign they have fully loaded and spacefinder can proceed
+					window.addEventListener('message', (event) => {
+						if (event.source !== iframe.contentWindow) return;
+
+						const message: Record<string, string | number> =
+							JSON.parse(event.data) as Record<
+								string,
+								string | number
+							>;
```

Comment:
If you've defined the type of message, you don't need to cast `JSON.parse`. The type of an unknown JSON is `any`, so it will match anything by default.

If you want to ensure this data is parsed adequately, you'll need a parser/type predicate function. You might be happy to just assume the shape of data, though.

Reply:
It may also be worthwhile wrapping the `JSON.parse` in a try-catch block, as anything that isn't parsable will [throw an exception](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#exceptions).

Reply:
Good point, have added @chrislomaxjones!